apps/src/code-studio/components/progress/CourseProgress.jsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import Radium from 'radium';
+import { connect } from 'react-redux';
+import _ from 'lodash';
+import experiments from '@cdo/apps/util/experiments';
+import { stageShape } from './types';
+import CourseProgressRow from './CourseProgressRow';
+import CourseOverviewTopRow from './CourseOverviewTopRow';
+import { ViewType } from '@cdo/apps/code-studio/stageLockRedux';
+import color from "@cdo/apps/util/color";
+import ProgressTable from '@cdo/apps/templates/progress/ProgressTable';
+
+const styles = {
+  flexHeader: {
+    padding: '8px 11px',
+    margin: '20px 0 0 0',
+    borderRadius: 5,
+    background: color.cyan,
+    color: color.white
+  }
+};
+
+/**
+ * Stage progress component used in level header and course overview.
+ */
+const CourseProgress = React.createClass({
+  propTypes: {
+    onOverviewPage: React.PropTypes.bool.isRequired,
+
+    // redux provided
+    perLevelProgress: React.PropTypes.object.isRequired,
+    scriptName: React.PropTypes.string.isRequired,
+    professionalLearningCourse: React.PropTypes.bool,
+    focusAreaPositions: React.PropTypes.arrayOf(React.PropTypes.number),
+    stages: React.PropTypes.arrayOf(stageShape),
+    peerReviewStage: stageShape,
+    viewAs: React.PropTypes.oneOf(Object.values(ViewType)).isRequired,
+  },
+
+  render() {
+    const {
+      stages,
+      peerReviewStage,
+      professionalLearningCourse,
+      focusAreaPositions,
+      scriptName,
+      viewAs
+    } = this.props;
+    const groups = _.groupBy(stages, stage => (stage.flex_category || 'Content'));
+    const showGroupHeaders = _.size(groups) > 1;
+    // Add an additional group for any peer reviews
+    if (peerReviewStage) {
+      // peerReviewStage.flex_category will always be "Peer Review" here
+      groups[peerReviewStage.flex_category] = [peerReviewStage];
+    }
+
+    let count = 1;
+
+    const hasLevelProgress = Object.keys(this.props.perLevelProgress).length > 0;
+
+    const progressRedesign = experiments.isEnabled('progressRedesign');
+
+    return (
+      <div>
+        {this.props.onOverviewPage && (
+          <CourseOverviewTopRow
+            professionalLearningCourse={professionalLearningCourse}
+            hasLevelProgress={hasLevelProgress}
+            scriptName={scriptName}
+            viewAs={viewAs}
+          />
+        )}
```

Comment:
I considered connecting `CourseOverviewTopRow` and having it not be a child of `CourseProgress`. However, because in the redesign it contains a toggle for switching between summary/detail view it felt like it maybe did belong here

![image](https://cloud.githubusercontent.com/assets/1767466/24467685/5d5869a6-146a-11e7-8681-6b166d3ae6b5.png)